.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+```
+/obj/item/device/cake
+    [cake code here]
+
+    proc/eat_cake()
+        [proc code here]
+```
+
+### qdel() and Destroy() usage
+All objects with an applicable type need to be deleted by `qdel()`, as opposed to the regular `del()` proc. While conducting this action, make sure you remove all possible references to the object you assign for deletion *after* calling `qdel()`. This will enable the ProcessScheduler controller garbage collector to handle the objects assigned to it at its own pace, thus reducing lag in the long run.
+
+An example of how to use `qdel()`:
+```
+/obj/item/plate
+	var/obj/item/cake/cake
+
+/obj/item/plate/New()
+	cake = New()
+
+// Eat the cake and destroy the cake object.
+/obj/item/plate/proc/eat_cake()
+	qdel(cake)	// Call qdel()
+	cake = null	// Set local reference to null to assist the GC.
+```
+
+The `Destroy()` proc for objects should be defined, if there are any special operations that need to be conducted when an object is assigned for destruction with `qdel()`. Normally, it would set all object references that that specific item may contain to null, and destroy them as necessary. It is important to know that the best case scenario for the garbage collector is this: an object passed to it should not reference, or be referenced by any other ingame object.
+
+Note that any modified `Destroy()` proc **must always return the original definition (`return ..()`) call!**
+
+An example of how to define `Destroy()` for an item that needs it:
+```
+/obj/item/plate
+	var/obj/item/cake/cake
+
+/obj/item/plate/New()
+	cake = New()
+
+/obj/item/plate/Destroy()
+	if (src.cake)	// We potentially have a reference.
+		qdel(cake)	// Delete the referenced item -- this doesn't always have to be done.
+		cake = null	// Set the pointer to null. This is the important bit.
+					// All pointers that the plate item contains are now null. This will speed up the GC.
+
+	return ..()		// Return the original definition of the proc.
+```
+
+`qdel()` is **not** capable of handling the following types of objects:
+* file
+* savefile
+* SQLLite object
+* Client object
+
+You will have to use the regular `del()` proc to delete any object of that type.
+
+### HTML styling for user output
+All text output to the user, specially if the output operator `<<` is used, should be formatted in proper HTML. DM text macros for styling, such as `\red` and `\blue`, are no longer to be used actively. This will enable the modification of used HTML styling later down the line, via the centralized .css files. It will also enable a switch from an output panel, to other output methods.
+
+For reference, here are the standard span classes for user output, and the correlation between them and the DM text macros:
+* `<span class="danger"></span>` corresponds to `\red` and is bold.
+* `<span class="warning"></span>` also corresponds to `\red` and is not bold.
+* `<span class="notice"></span>` corresponds to `\blue` and is not bold.
```

Comment:
It would be good to mention where the central style sheet is in the code. I was amazed when someone eventually pointed me to it, we have so many more span classes than just these three